### PR TITLE
Update macOS buildenv variables

### PR DESCRIPTION
**Pull request by olvap377**
For issue #9806

extra_configure_options:
    8: '--with-toolchain-type=clang'

SDKPATH removed

node_labels changed to sw.os.osx.10_14
  build_env var changed to MACOSX_DEPLOYMENT_TARGET=10.10.0

@jdekonin

### CODE REVIEW

**Comment by jdekonin** [comment]
jenkins test sanity osx jdk8,jdk11

File: buildenv/jenkins/variables/defaults.yml [hunk 1]
```diff
@@ -545,17 +545,13 @@ x86-64_mac:
...
   node_labels:
-    build:
-      all: 'ci.role.build && hw.arch.x86 && sw.os.osx.10_13'
-      8: 'ci.role.build && hw.arch.x86 && sw.os.osx.10_11'
+    build: 'ci.role.build && hw.arch.x86 && sw.os.osx.10_14'
   build_env:
-    vars:
-      all: 'OPENJ9_JAVA_OPTIONS=-Xdump:system+java:events=systhrow,filter=java/lang/ClassCastException,request=exclusive+prepwalk+preempt'
-      8: 'MACOSX_DEPLOYMENT_TARGET=10.9.0 SDKPATH=/Users/jenkins/Xcode4/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk'
+    vars: 'MACOSX_DEPLOYMENT_TARGET=10.10.0,OPENJ9_JAVA_OPTIONS=-Xdump:system+java:events=systhrow,filter=java/lang/ClassCastException,request=exclusive+prepwalk+preempt'
```
Comment (jdekonin) [inline buildenv/jenkins/variables/defaults.yml:554]: Previous line was space separated options.

Comment (olvap377) [reply buildenv/jenkins/variables/defaults.yml]: Fixed

**Comment by jdekonin** [comment]
jenkins test sanity osx jdk8,jdk11

**Comment by jdekonin** [comment]
Failed due to missing build label on the 1-.14 systems.  I have opened a PR to address that; https://github.com/eclipse-cbi/jiro/pull/121

**Comment by olvap377** [comment]
@jdekonin PR merged [eclipse-cbi/jiro#121](https://github.com/eclipse-cbi/jiro/pull/121)

**Comment by jdekonin** [comment]
jenkins test sanity osx jdk8,jdk11

**Comment by jdekonin** [comment]
JDK8 compile happens on osx1014-x86-2 and testing runs successfully on osx1011-x86-1
JDK11 compile happens on osx1014-x86-1 and testing runs successfully on osx1013-x86-4

I believe this is ready to merge  @pshipton @DanHeidinga @keithc-ca

File: buildenv/jenkins/variables/defaults.yml [hunk 2]
```diff
@@ -545,17 +545,13 @@ x86-64_mac:
...
   node_labels:
-    build:
-      all: 'ci.role.build && hw.arch.x86 && sw.os.osx.10_13'
-      8: 'ci.role.build && hw.arch.x86 && sw.os.osx.10_11'
+    build: 'ci.role.build && hw.arch.x86 && sw.os.osx.10_14'
   build_env:
-    vars:
-      all: 'OPENJ9_JAVA_OPTIONS=-Xdump:system+java:events=systhrow,filter=java/lang/ClassCastException,request=exclusive+prepwalk+preempt'
-      8: 'MACOSX_DEPLOYMENT_TARGET=10.9.0 SDKPATH=/Users/jenkins/Xcode4/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk'
+    vars: 'MACOSX_DEPLOYMENT_TARGET=10.10.0 OPENJ9_JAVA_OPTIONS=-Xdump:system+java:events=systhrow,filter=java/lang/ClassCastException,request=exclusive+prepwalk+preempt'
```
Comment (keithc-ca) [inline buildenv/jenkins/variables/defaults.yml:554]: Why is the deployment target updated from 10.9.0 to 10.10.0? I think it should just be removed; I don't think it has any effect anyway - build and configure scripts appear to ignore it.

Comment (jdekonin) [reply buildenv/jenkins/variables/defaults.yml:554]: Its not used by xcode/clang to support older releases?

Comment (keithc-ca) [reply buildenv/jenkins/variables/defaults.yml:554]: Even if it is used by the compiler, the value supplied here is ignored; see e.g. https://github.com/ibmruntimes/openj9-openjdk-jdk11/blob/openj9/closed/autoconf/custom-spec.gmk.in#L109.

Comment (jdekonin) [reply buildenv/jenkins/variables/defaults.yml:554]: Right and set in https://github.com/ibmruntimes/openj9-openjdk-jdk11/blob/22c634c40143ed77119efa3800a8a9f1a726d19f/make/autoconf/flags.m4#L119 to 10.9.0, understood.  So back to you @olvap377 to remove `MACOSX_DEPLOYMENT_TARGET=10.10.0 `

Comment (olvap377) [reply buildenv/jenkins/variables/defaults.yml]: Removed

**Comment by keithc-ca** [comment]
Jenkins test sanity osx jdk8,jdk11

**Comment by keithc-ca** [comment]
The 'failure' on jdk8 makes no sense, but it's not a new issue (see #6361):
```
Running test cmdLineTester_CryptoTest_0 ...
===============================================
cmdLineTester_CryptoTest_0 Start Time: Tue Aug 11 16:13:14 2020 Epoch Time (ms): 1597176794543
variation: NoOptions
JVM_OPTIONS:  
*** Starting test suite: OpenSSL Tests ***
Testing: CryptoTest
Test start time: 2020/08/11 16:13:14 Eastern Standard Time
Running command: "/Users/jenkins/workspace/Test_openjdk8_j9_sanity.functional_x86-64_mac_Personal_testList_0/openjdkbinary/j2sdk-image/bin/java"   -Djdk.nativeCryptoTrace=true -cp "/Users/jenkins/workspace/Test_openjdk8_j9_sanity.functional_x86-64_mac_Personal_testList_0/openjdk-tests/TKG/../../jvmtest/functional/cmdLineTests/openssl/"openssltest.jar org.openj9.test.openssl.CryptoTest
Time spent starting: 40 milliseconds
Time spent executing: 318 milliseconds
Test result: PASSED
Output from test:
 [OUT] Crypto test COMPLETED
 [ERR] MessageDigest load - using Native crypto library.
 [ERR] CipherCore Load - using native crypto library.
>> Success condition was found: [Output match: using native crypto library]
>> Required condition was found: [Output match: Crypto test COMPLETED]
>> Failure condition was not found: [Output match: Native crypto library load failed]
>> Failure condition was not found: [Output match: Crypto test FAILED]


---TEST RESULTS---
Number of PASSED tests: 1 out of 1
Number of FAILED tests: 0 out of 1

cmdLineTester_CryptoTest_0_FAILED
```

**Pull request merged**